### PR TITLE
Add support to load face from memory file-like object

### LINKED ISSUE #32: Possible to create a Font object from memory?
A binding to `FT_New_Memory_Face` for creating a Font object. I have a binary font in memory and I don't want to save it to disk and load it again.

Is it possible to do so? 
The Font constructor in the docs seems to only take a filename.

### PR DESCRIPTION
**Pull request by yifeikong**
close #32

### CODE REVIEW

File: freetype/__init__.py [hunk 1]
```diff
@@ -977,17 +977,25 @@ def __init__( self, filename, index = 0 ):
...
-            error = FT_New_Face( library, u_filename, index, byref(face) )
-        except UnicodeError:
-            with open(filename, mode='rb') as f:
-                filebody = f.read()
+        if hasattr(path_or_file, 'read'):
+            filebody = path_or_file.read()
             error = FT_New_Memory_Face( library, filebody, len(filebody),
                                         index, byref(face) )
             self._filebodys.append(filebody)  # prevent gc
+            self._filename = ":memory:"
```
Comment (adrientetar) [inline freetype/__init__.py]: Could probably just make it None here

Comment (yifeikong) [reply freetype/__init__.py]: sqlite3 uses `:memory:` to denote a memory file, I thought this make sense.

Comment (madig) [reply freetype/__init__.py]: Using `None`, you could in code later use `if not x.filename: save or something`. Although I'm not a fan of assigning `None` to anything other than using if for default arguments, so assigning an empty string `""` would be even nicer. Unless the code at some point wants to switch to using pathlib.Path objects for all path related things.

File: freetype/__init__.py [hunk 2]
```diff
@@ -977,17 +977,25 @@ def __init__( self, filename, index = 0 ):
...
+        else:
+            filename = path_or_file
+            try:
+                u_filename = c_char_p(_encode_filename(filename))
+                error = FT_New_Face( library, u_filename, index, byref(face) )
+            except UnicodeError:
```
Comment (madig) [inline freetype/__init__.py]: Hm. Would it make sense to just always forego FT_New_Face and use FT_New_Memory_Face instead and let Python handle the opening? One code path less.

**Pull request closed without merging**